### PR TITLE
50 add group model

### PR DESCRIPTION
**Pull request by kalebdavis**
closes PagerNation/PagerNation#50

### CODE REVIEW

File: server/models/group.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * id - default id given by mongodb
+ * users - list of user objects that are included in the group
+ * escalationPolicy - escalation policy for this group
+ */
+
+const GroupSchema = new mongoose.Schema({
+  users: [mongoose.Schema.Types.ObjectId],
+  escalationPolicy: {
+    type: mongoose.Schema.ObjectId,
```
Comment (jarrydlee) [inline server/models/group.js]: This doesn't actually embed the document, it just adds the `ObjectId` that can get the document once you call the `.populate('escalationPolicy')` method when you're getting the group object.

**Comment by jarrydlee** [comment]
What's the status on this?

**Comment by kalebdavis** [comment]
Ready for review

File: server/tests/models/group.test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { expect } from 'chai';
+import Group from '../../models/group';
+import User from '../../models/user';
+import EscalationPolicy from '../../models/escalationPolicy';
+
+describe('Groups', function() {
+  let subscriberRotationInterval = 7;
```
Comment (jarrydlee) [inline server/tests/models/group.test.js]: nit: Ideally, const should be used if we aren't going to change the variable

**Comment by jarrydlee** [comment]
Apart from that, lgtm

File: server/models/group.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+ * Group Schema
+ *
+ * id - default id given by mongodb
+ * users - list of user objects that are included in the group
+ * escalationPolicy - escalation policy for this group
+ */
+
+const GroupSchema = new mongoose.Schema({
+  users: [mongoose.Schema.Types.ObjectId],
+  escalationPolicy: EscalationPolicySchema
```
Comment (paradime) [inline server/models/group.js]: if required, it should say required. if not, it should default to null

File: server/tests/models/group.test.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
```
Comment (kalebdavis) [inline server/tests/models/group.test.js]: I'll leave that up to you

**Comment by jarrydlee** [comment]
Can you change the ```function() {}``` to ```() -> {}```

**Comment by kalebdavis** [comment]
What's the difference between the two? Can you point me towards some docs?

**Comment by jarrydlee** [comment]
[Here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) and [here](https://www.sitepoint.com/es6-arrow-functions-new-fat-concise-syntax-javascript/) are some good examples of what it offers.

TL;DR: ES6 anonymous functions, there are some small differences, but mostly the same.

File: server/tests/models/group.test.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    const groupData = {
+      users: [userObjectId],
+      escalationPolicy: escPolicy
+    }
+
+    const group = new Group(groupData);
+
+    it('should create a new group', (done) => {
+      group.save((err, group) => {
+        if(err) {
```
Comment (paradime) [inline server/tests/models/group.test.js]: I like where you're going here. There are two approaches that I would just clean this up. According to [Mocha](mochajs.org), you can do this in one line,
`if (err) done(err);` 

However, in other tests, we've just been asserting that the error doesn't exist.
`expect(err).to.not.exist;`

Personally, I like chai doing the assertions, however, the asertion doesn't give you the reason why it failed.

**Comment by paradime** [comment]
Straight 🔥 Nice tests!

File: server/tests/models/group.test.js (hunk 5 above)
Comment (jarrydlee) [inline server/tests/models/group.test.js]: Just because you can do it in one line, doesn't mean that you should. For the amount of space it saves, I think that `if (err) done(err);` is a code smell.

I agree with using `expect(err).to.not.exist` here
